aten/src/ATen/templates/RegisterDispatchKey.cpp

```diff
@@ -5,12 +5,19 @@
 #define __STDC_FORMAT_MACROS
 #endif
 
+#if defined(CAFFE2_BUILD_MAIN_LIB)        || \
+    defined(TORCH_CUDA_BUILD_MAIN_LIB)    || \
+    defined(TORCH_HIP_BUILD_MAIN_LIB)     || \
+    defined(TORCH_CUDA_CU_BUILD_MAIN_LIB) || \
+    defined(TORCH_CUDA_CPP_BUILD_MAIN_LIB)
+#define TORCH_ASSERT_ONLY_METHOD_OPERATORS
```

Comment:
Where does this not work? should we leave a comment here for the next person who's going to try and fix it?
(sorry if this is in a PR up the stack, I'm reviewing on the way up)

Reply:
This special case only excludes external projects such as XLA which re-use some of pytorch's codegen, including `RegisterDispatchKey.cpp`.